### ISSUE TITLE
Show step validation fails if using a showtoken to set the 'fade' value of a light. 

### ISSUE THREAD

**Issue by kaydeeH** [missionpinball/mpf#1186]
Take the following example config:
```
show_player:
  some_event:
    some_show:
      speed: 1
      loops: -1
      priority: 100
      show_tokens:
        my_token: 500ms

shows:
  some_show:
    -duration: 1
      lights:
        l_some_light:
          color: FFFFFF
          fade: (my_token)
```
When you attempt to run, you get the following stack trace:
```
2018-06-03 14:12:20,471 : ERROR : asyncio : Exception in callback AsyncioSyncAssetManager._done(<Task finishe... '(STUFF)'",)>) at c:\program files\python35\lib\site-packages\mpf\core\assets.py:617
handle: <Handle AsyncioSyncAssetManager._done(<Task finishe... '(STUFF)'",)>) at c:\program files\python35\lib\site-packages\mpf\core\assets.py:617>
Traceback (most recent call last):
  File "c:\program files\python35\lib\asyncio\events.py", line 127, in _run
    self._callback(*self._args)
  File "c:\program files\python35\lib\site-packages\mpf\core\assets.py", line 623, in _done
    future.result()
  File "c:\program files\python35\lib\asyncio\futures.py", line 294, in result
    raise self._exception
  File "c:\program files\python35\lib\asyncio\tasks.py", line 240, in _step
    result = coro.send(None)
  File "c:\program files\python35\lib\asyncio\coroutines.py", line 210, in coro
    res = func(*args, **kw)
  File "c:\program files\python35\lib\site-packages\mpf\core\assets.py", line 605, in wait_for_asset_load
    result = self._load_sync(asset)
  File "c:\program files\python35\lib\site-packages\mpf\core\assets.py", line 597, in _load_sync
    asset.do_load()
  File "c:\program files\python35\lib\site-packages\mpf\assets\show.py", line 83, in do_load
    self._do_load_show(None)
  File "c:\program files\python35\lib\site-packages\mpf\assets\show.py", line 167, in _do_load_show
    self._process_step_actions(step, actions)
  File "c:\program files\python35\lib\site-packages\mpf\assets\show.py", line 197, in _process_step_actions
    actions[key] = self.machine.show_controller.show_players[key].validate_config_entry(value, self.name)
  File "c:\program files\python35\lib\site-packages\mpf\config_players\device_config_player.py", line 30, in validate_config_entry
    self._validate_config_item(device, device_settings))
  File "c:\program files\python35\lib\site-packages\mpf\config_players\device_config_player.py", line 56, in _validate_config_item
    device_settings = self._parse_config(device_settings, device)
  File "c:\program files\python35\lib\site-packages\mpf\core\config_player.py", line 126, in _parse_config
    return self.get_full_config(config)
  File "c:\program files\python35\lib\site-packages\mpf\config_players\light_player.py", line 130, in get_full_config
    super().get_full_config(value)
  File "c:\program files\python35\lib\site-packages\mpf\core\config_player.py", line 194, in get_full_config
    self.config_file_section, value, base_spec='config_player_common')
  File "c:\program files\python35\lib\site-packages\mpf\core\config_validator.py", line 169, in validate_config
    validation_failure_info=(validation_failure_info, k))
  File "c:\program files\python35\lib\site-packages\mpf\core\config_validator.py", line 208, in validate_config_item
    validation_failure_info)
  File "c:\program files\python35\lib\site-packages\mpf\core\config_validator.py", line 596, in validate_item
    return self.validator_list[validator](item, validation_failure_info=validation_failure_info)
  File "c:\program files\python35\lib\site-packages\mpf\core\config_validator.py", line 491, in _validate_type_ms
    return Util.string_to_ms(item)
  File "c:\program files\python35\lib\site-packages\mpf\core\utility_functions.py", line 553, in string_to_ms
    return int(time_string)
ValueError: invalid literal for int() with base 10: '(MY_TOKEN)'
```

I dug in a bit and I believe this is happening because the show.py file doesn't call `self._get_tokens()` until after `self._process_step_actions(step, actions)`. For string fields, this won't matter, but for anything that's evaluated and is expecting a non-string, validation fails. I suspect if you simply call the token replacement prior to validation, it should work on a properly formatted file, but I suppose the token replacement may go nuts if the structure is invalid.

TLDR> Show token replacement will cause a crash on any non-string value.

**Comment by jabdoa2** [comment]
Good find. Eventually, we need to see if we can replace show token with placeholders or if we need two different concepts. I guess a int_or_token validatior. Would be the quick fix here
